make-base/docker-compose.mk

```diff
@@ -1,68 +1,45 @@
 export COMPOSE_PROJECT_NAME:=monoetna
-COMPOSE_MIXINS:=docker-compose.shared.yml $(COMPOSE_MIXINS)
 app_service_name:=${app_name}_app
-baseTag:=$(shell basename "$$(pwd)")
-fullTag:=$(IMAGES_PREFIX)$(baseTag)$(IMAGES_POSTFIX)
-baseFeTag:=${app_name}_app_fe
-fullFeTag:=$(IMAGES_PREFIX)$(baseFeTag)$(IMAGES_POSTFIX)
 containerSh:=bash
 
-docker-compose.yml:: $(wildcard ../docker/*.shared.yml) ../docker/default_compose
-	COMPOSE_MIXINS="$(COMPOSE_MIXINS)" ../docker/default_compose docker-compose.yml
-
 .PHONY: images
-images: config-ready docker-compose.yml
-	@ make -C ../docker build
 
-config-ready:: docker-compose.yml
+ifneq ($(wildcard ./docker-compose.yml),)
+$(foreach df,$(call buildable_dependent_compose_image_dockerfiles,$(shell dirname $(firstword $(MAKEFILE_LIST)))/docker-compose.yml),$(eval $(call image_target2,$(shell dirname $(df)),$(call buildable_dependent_image_dockerfiles,$(df)))))
+images: config-ready $(foreach df,$(call buildable_dependent_compose_image_dockerfiles,$(shell dirname $(firstword $(MAKEFILE_LIST)))/docker-compose.yml),$(shell dirname $(df))/image.marker)
+endif
+
+# docker-compose.yml is a file that must be hand written by the developer in any
+# project that uses docker-compose to manage its dev environment.
+config-ready:: docker-compose.yml .dockerignore
 	@ true
 
-docker-ready:: images
+compose-ready:: docker-compose.yml images
 	@ true
 
-up:: docker-ready
+up:: compose-ready
 	@ docker-compose up -d
 
 down:: docker-compose.yml
 	@ docker-compose down
 
-ps::
+ps:: compose-ready
 	@ docker-compose ps
 
-restart:: docker-ready
+restart:: compose-ready
 	@ docker-compose restart
 
-bash:: docker-ready
+bash:: compose-ready
 	@ docker-compose run --rm $(app_service_name) $(containerSh)
 
-logs::
+logs:: compose-ready
 	@ docker-compose logs -f
 
-test:: docker-ready
-	@ true
+run-image-test:: compose-ready
+	@ docker-compose up
 
 .dockerignore:
-	if [ -e Dockerfile]; then cp ../docker/.dockerignore.template ./.dockerignore; fi
-
-release-build:: .dockerignore
-	if [ -e Dockerfile ]; then touch /tmp/.release-test; ../docker/build_image Dockerfile $(BUILD_REQS) -- $(BUILD_ARGS); else touch /tmp/etna-build-markers/$(baseTag); fi
-	if [ -e $(baseFeTag)/Dockerfile ]; then ../docker/build_image $(baseFeTag)/Dockerfile -- $(BUILD_ARGS); fi
-
-/tmp/.release-test: /tmp/etna-build-markers/$(baseTag)
```

Comment:
Here is the problematic insanity I was doing before that broke archimedes builds.
What I was doing before was using adhoc `marker` files to track build processes and tests in order to prevent extra work from happening in CI.
The problem is that these markers were not scoped well and had unintended side effects when multiple projects built together (archimedes alongside vulcan).

The new approach removes all this custom logic and replaces it with logic from `utils.mk` that is consistent and scoped globally and correctly so as to prevent this state mix up.